config/config.go

```diff
@@ -170,3 +173,192 @@ func defaultRunPath() string {
+	re := regexp.MustCompile(`[^\."']+|"([^"]*)"|'([^']*)`)
+	pathParts := re.FindAllString(path, -1)
+
+	for i := 0; i < len(pathParts); i++ {
+		// If we did have a quoted string we need to prune it
+		pathParts[i] = strings.Trim(pathParts[i], `"`)
+	}
+
+	return pathParts
+}
+
+func validatePath(t reflect.Type, path []string) error {
+	if len(path) == 0 {
+		return nil
+	}
+
+	if t.Kind() == reflect.Map {
+		// Maps are only for git repos. Grab the kind on the other side of the map
+		// and advance; to validate the path on the other side of the repo name
+		return validatePath(t.Elem(), path[1:])
+	}
+
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		tag := field.Tag.Get("yaml")
+
+		if tag == path[0] {
+			return validatePath(field.Type, path[1:])
+		}
+	}
+
+	return fmt.Errorf("no path for %s", strings.Join(path, "."))
+}
+
+func valueToYaml(value string) (*yaml.Node, error) {
+	valueNode := &yaml.Node{}
+	if err := yaml.Unmarshal([]byte(value), valueNode); err != nil {
+		return nil, fmt.Errorf("%s is not a valid YAML value", value)
+	}
+
+	// Unfold all the yaml so its not mixed inline and flow styles in the final document
+	var fixStyling func(node *yaml.Node)
+	fixStyling = func(node *yaml.Node) {
+		node.Style = 0
+
+		for _, n := range node.Content {
+			fixStyling(n)
+		}
+	}
+	fixStyling(valueNode)
+
+	return valueNode.Content[0], nil
+}
+
+func pathToYaml(path []string, value *yaml.Node) []*yaml.Node {
+	yamlNodes := []*yaml.Node{}
+
+	var last *yaml.Node
+
+	for i, seg := range path {
+		key := &yaml.Node{
+			Kind:  yaml.ScalarNode,
+			Value: seg,
+		}
+
+		mapping := &yaml.Node{
+			Kind: yaml.MappingNode,
+		}
+
+		if i == len(path)-1 {
+			// Last node should assign path as the value, not another mapping node
+			// Otherwise we would need to dig it up again.
+
+			if last == nil {
+				// Single depth special case
+				yamlNodes = append(yamlNodes, key, value)
+				continue
+			}
+
+			last.Content = append(last.Content, key, value)
+		} else if last == nil {
+			// First, top level mapping node
+			yamlNodes = append(yamlNodes, key, mapping)
+			last = mapping
+		} else {
+			// Middle of the road regular case
+			last.Content = append(last.Content, key, mapping)
+			last = mapping
+		}
+	}
+
+	return yamlNodes
+}
+
+func setYamlValue(node *yaml.Node, path []string, value *yaml.Node) []string {
+	switch node.Kind {
+	case yaml.DocumentNode:
+		for _, c := range node.Content {
+			path = setYamlValue(c, path, value)
+		}
+
+	case yaml.MappingNode:
+		for i := 0; i < len(node.Content); i += 2 {
+			// Keys/Values are inline. Count by twos to get it right.
+			key := node.Content[i]
+			val := node.Content[i+1]
+
+			if len(path) > 0 && key.Value == path[0] {
+				path = path[1:]
+
+				if len(path) == 0 {
+					node.Content[i+1] = value
+					return []string{}
+				}
+
+				path = setYamlValue(val, path, value)
+			}
+		}
+
+	default: // Sequence, Scalar nodes get skipped
```

Comment:
As well as parse errors; I think I cleaned up all empty/zero documents getting in here?